api/utils/pingconn/tls.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pingconn
+
+import (
+	"crypto/tls"
+
+	"github.com/gravitational/trace"
+)
+
+// NewTLS returns a ping connection wrapping the provided tls.Conn.
+func NewTLS(conn *tls.Conn) *PingTLSConn {
+	return &PingTLSConn{
+		Conn: conn,
+		ping: New(conn),
+	}
+}
+
+// PingTLSConn wraps a tls.Conn and adds ping capabilities to it.
+type PingTLSConn struct {
+	*tls.Conn
+
+	ping *PingConn
+}
```

Comment:
From the `pingconn` perspective we don't care if this is a TLS conn or new.Conn or other type. 
This matters only in scenario where a consumers need to distinguish the connection type. 

Wonder if we can just switch to net.Conn in our PingConn implementation and  
convert to pingConnTLS in our proxy implementation like: 
```go
type pingConnTLS struct {
	utils.TLSConn
	pingconn.PingConn
}

func ToTLSConn(pconn pingconn.PingConn) (utils.TLSConn, error) {
	if tlsConn, ok := pconn.Conn.(*tls.Conn); ok {
		return &pingConnTLS{
			PingConn: pconn,
			TLSConn:  tlsConn,
		}, nil
	}
	return nil, trace.BadParameter("Unexpected connection type: %T", pconn.Conn)
}
```